plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2554,7 +2519,7 @@ private void cleanExtraOutputFiles(HdfsContext hdfsContext, String queryId, Stri
             while (filesToDeleteIterator.hasNext()) {
                 String fileName = filesToDeleteIterator.next();
                 LOG.debug("Deleting failed attempt file %s/%s for query %s", location, fileName, queryId);
-                fileSystem.delete(new Path(location, fileName), false);
+                fileSystem.deleteFile(new Path(location, fileName).toString());
```

Comment:
We have now in `TrinoFileSystem` the method `deleteFiles` which performs the deletion in batches on S3. It would probably be a good fit here.

The files to be deleted area already in a in-memory queue, so we can safely them  to this method.